homeassistant/helpers/update_coordinator.py

```diff
@@ -84,9 +84,18 @@ def __init__(
         self.update_interval = update_interval
         self._shutdown_requested = False
         if config_entry is UNDEFINED:
+            # late import to avoid circular imports
+            from . import frame  # pylint: disable=import-outside-toplevel
+
+            frame.report_usage(
+                "rely on the ContextVar, but should passing the config entry explicitly - see "
+                "https://developers.home-assistant.io/blog/xxxxxxx",
+                core_behavior=frame.ReportBehavior.ERROR,
+                custom_integration_behavior=frame.ReportBehavior.LOG,
+                breaks_in_ha_version="2026.3",
```

Comment:
Should not be there as it should not break https://github.com/home-assistant/core/pull/138161/files#r1948549150.
Probably good to ensure that in a test, so it does not get cleaned up later.